ggshield/cmd/hmsl/check_secret_manager/__init__.py

```diff
@@ -14,6 +14,40 @@
     },
     hidden=True,
 )
+@click.option(
+    "--use-cli-token",
+    "use_cli_token",
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help="Instead of getting the token from the an environment variable, "
+    "get the token from the CLI tool.",
+)
+@click.option(
+    "--url",
+    "url",
+    required=False,
+    type=str,
+    help="The URL of the secret manager server.",
+)
+@click.option(
+    "--recursive",
+    "-r",
+    "recursive",
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help="If the secret manager path is a directory and not a file, explore recursively.",
+)
```

Comment:
It's not a good idea to define the options at the group level, because Click has the annoying limitation of not propagating group options down to individual commands. This means that with the current code, you **must** pass the URL like this:

```
ggshield hmsl check-secret-manager --url https://example.com hashicorp-vault
```

If you define the URL later, like that:


```
ggshield hmsl check-secret-manager hashicorp-vault --url https://example.com
```

then it won't work :(

It's better to define the options at the command level, and factorize them in a common_options module later if necessary (see secret_scan_common_options.py for example)